fish-rust/src/common.rs

```diff
@@ -1177,7 +1179,7 @@ pub fn cstr2wcstring(input: &[u8]) -> WString {
     str2wcstring(&input[0..strlen])
 }
 
-pub fn charptr2wcstring(input: *const libc::c_char) -> WString {
+pub unsafe fn charptr2wcstring(input: *const libc::c_char) -> WString {
```

Comment:
I think we can avoid the need for `unsafe` by using `pub(crate)` instead of `pub` (preserving existing meaning)